org.openhab.binding.zigbee.ember/ESH-INF/thing/controller_ember.xml

```diff
@@ -8,100 +8,160 @@
+				<label>Channel</label>
+				<description>Channel number</description>
+				<default>-1</default>
+				<options>
+					<option value="-1">Auto</option>
+					<option value="11">Channel 11</option>
+					<option value="12">Channel 12</option>
+					<option value="13">Channel 13</option>
+					<option value="14">Channel 14</option>
+					<option value="15">Channel 15</option>
+					<option value="16">Channel 16</option>
+					<option value="17">Channel 17</option>
+					<option value="18">Channel 18</option>
+					<option value="19">Channel 19</option>
+					<option value="20">Channel 20</option>
+					<option value="21">Channel 21</option>
+					<option value="22">Channel 22</option>
+					<option value="23">Channel 23</option>
+					<option value="24">Channel 24</option>
+					<option value="25">Channel 25</option>
+				</options>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="zigbee_panid" type="Integer" groupName="network" required="false">
+				<label>PAN ID</label>
+				<description>PAN Network ID</description>
+				<default>0</default>
+				<options>
+					<option value="65535">Auto</option>
+				</options>
+				<limitToOptions>false</limitToOptions>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="zigbee_extendedpanid" type="text" groupName="network" required="false">
+				<label>Extended PAN ID</label>
+				<description>Extended PAN Network ID: 16 byte hexadecimal value</description>
+				<default>0000000000000000</default>
+				<advanced>true</advanced>
+			</parameter>
+
+			<parameter name="zigbee_networkkey" type="text" groupName="network">
+				<label>Network Security Key</label>
+				<description>Set the network security key</description>
+				<default>00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00</default>
+				<advanced>true</advanced>
+			</parameter>
+
+		</config-description>
+
 	</bridge-type>
-	
+
+	<channel-type id="ember_ash_rx_dat">
+		<item-type>Number</item-type>
+		<label>Data frames received</label>
+		<description>Provides a count of DATA frames received by the host.</description>
+		<state readOnly="true">
+		</state>
+	</channel-type>
+
+	<channel-type id="ember_ash_tx_dat">
+		<item-type>Number</item-type>
+		<label>Data frames sent</label>
+		<description>Provides a count of DATA frames sent by the host.</description>
+		<state readOnly="true">
+		</state>
+	</channel-type>
+
+	<channel-type id="ember_ash_rx_nak">
+		<item-type>Number</item-type>
+		<label>NAK frames received</label>
+		<description>Provides a count of DATA frame with an error or that were discarded by the NCP due to lack of memory.
+		</description>
+		<state readOnly="true">
+		</state>
+	</channel-type>
+
+	<channel-type id="ember_ash_tx_nak">
+		<item-type>Number</item-type>
+		<label>NAK frames sent</label>
+		<description>Provides a count of DATA frame with an error or that were discarded by the host due to lack of memory.
```

Comment:
Question: Why is this "due to lack of memory", and not also due to other issues? Background of the question: I had a look into the `AshFrameHandler`, which will (as far as I see) also send a NAK if it receives a bad package. Or am I thinking in the wrong direction here?

Reply:
> Why is this "due to lack of memory"

That's just what the doc states. It might also be other reasons but the Silabs doc states the following -:

> • Purpose: Indicates receipt of a DATA frame with an error or that was discarded due to lack of memory.

Reply:
Doesn't the comment that you are are citing refer to the `NAK`s sent by the NCP to the host, i.e., to those `NAK`s counted in the channel `ember_ash_rx_nak`?

I don't understand yet why this also applies to the `NAK`s sent by the host to the NCP - as far as I see, the host sending a `NAK` to the NCP is unrelated to lack-of-memory issues, but rather to [bad backets](https://github.com/zsmartsystems/com.zsmartsystems.zigbee/blob/e94d9d907b9c10b4e8c7ca4ea0cdab195241606e/com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java#L169) and [out-of-sequence transmissions](https://github.com/zsmartsystems/com.zsmartsystems.zigbee/blob/e94d9d907b9c10b4e8c7ca4ea0cdab195241606e/com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java#L211).